onnx/defs/nn/defs.cc

```diff
@@ -1398,4 +1398,96 @@ ONNX_OPERATOR_SET_SCHEMA(
         .SetDoc(LRN_ver1_doc)
         .TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput));
 
+static const char* NonMaxSuppression_doc = R"DOC(
+Pruning away boxes that have high intersection-over-union (IOU) overlap with previously selected boxes.
+Bounding boxes with score less than score_threshold are removed. Bounding boxes are supplied as [y1, x1, y2, x2],
+where (y1, x1) and (y2, x2) are the coordinates of any diagonal pair of box corners and the coordinates
+can be provided as normalized (i.e., lying in the interval [0, 1]) or absolute.
+Note that this algorithm is agnostic to where the origin is in the coordinate system and more generally
+is invariant to orthogonal transformations and translations of the coordinate system;
+thus translating or reflections of the coordinate system result in the same boxes being selected by the algorithm.
+The output of this operation is a set of integers indexing into the input collection of bounding boxes representing the selected boxes.
+The bounding box coordinates corresponding to the selected indices can then be obtained using the gather operation.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    NonMaxSuppression,
+    9,
+    OpSchema()
+        .Attr(
+            "max_output_size",
+            "Integer representing the maximum number of boxes to be selected by non max suppression.",
+            AttributeProto::INT)
+        .Attr(
+            "iou_threshold",
+            "Float representing the threshold for deciding whether boxes overlap too much with respect to IOU. Value range [0, 1]. The default is 0.0",
+            AttributeProto::FLOAT,
+            0.0f)
+        .Attr(
+            "score_threshold",
+            "Float tensor representing the threshold for deciding when to remove boxes based on score.",
+            AttributeProto::FLOAT)
+        .Attr(
+            "pad_to_max_output_size",
+            "Optional. 1(true) - the output selected_indices is padded to be of length max_output_size. Defaults to 0(false).",
+            AttributeProto::INT,
+            OPTIONAL)
```

Comment:
put default value here please.

Reply:
I didn't find a overload Attr which can accept the (..., bool required, T defaultvalue)

---
In reply to: [242298590](https://github.com/onnx/onnx/pull/1695#discussion_r242298590) [](ancestors = 242298590)